firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -331,6 +334,35 @@ public void onOpen(SQLiteDatabase db) {
     }
   }
 
+  /**
+   * A simple queue that executes tasks in parallel on the Android's AsyncTask.THREAD_POOL_EXECUTOR
+   * and supports blocking on their completion.
+   *
+   * <p>This class is not thread-safe. In particular, `execute()` and `drain()` should not be called
+   * from parallel threads.
+   */
+  static class BackgroundQueue implements Executor {
```

Comment:
nit: This particular class in which you've placed this class is huge. I'd love to see us move things out of here if possible.

Why not promote this to top-level? It's also totally generic, so you could also dump it in `util`, but leaving it `local` seems fine too.

Reply:
Moved to a top-level class in Util. For now, it doesn't have tests though.